packages/server-wallet/src/objectives/objective-manager.ts

```diff
@@ -0,0 +1,197 @@
+              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+              await this.chainService.concludeAndWithdraw([protocolState.app.supported!]);
+              return;
+            case 'RequestLedgerFunding': {
+              const ledgerChannelId = await determineWhichLedgerToUse(protocolState.app, tx);
+              await LedgerRequest.requestLedgerFunding(
+                protocolState.app.channelId,
+                ledgerChannelId,
+                tx
+              );
+              return;
+            }
+            case 'RequestLedgerDefunding': {
+              const ledgerChannelId = await determineWhichLedgerToUse(protocolState.app, tx);
+              await LedgerRequest.requestLedgerDefunding(
+                protocolState.app.channelId,
+                ledgerChannelId,
+                tx
+              );
+              return;
+            }
+            default:
+              throw 'Unimplemented';
+          }
+        };
+
+        const nextAction = recordFunctionMetrics(executeProtocol(), this.timingMetrics);
+
+        if (nextAction) {
+          try {
+            await doAction(nextAction);
+          } catch (error) {
+            this.logger.error({error}, 'Error handling action');
+            await tx.rollback(error);
+            maybeError = error;
+            attemptAnotherProtocolStep = false;
+          }
+        } else {
+          channelResults.push(ChannelState.toChannelResult(protocolState.app));
+          attemptAnotherProtocolStep = false;
+        }
+      });
+    }
+
+    return {channelResults, outbox, error: maybeError};
+  }
+}
+
+// todo: refactor this away
+const createOutboxFor = (
+  channelId: Bytes32,
+  myIndex: number,
+  participants: Participant[],
+  data: Payload
+): Outgoing[] =>
+  participants
+    .filter((_p, i: number): boolean => i !== myIndex)
+    .map(({participantId: recipient}) => ({
+      method: 'MessageQueued' as const,
+      params: serializeMessage(data, recipient, participants[myIndex].participantId, channelId),
+    }));
+
+const determineWhichLedgerToUse = async (
+  channel: ChannelState.ChannelState,
+  txOrKnex: TransactionOrKnex
+): Promise<Bytes32> => {
+  if (channel?.supported) {
+    if (channel.fundingStrategy === 'Ledger') {
+      if (channel.fundingLedgerChannelId) {
+        const ledgerChannelId = channel.fundingLedgerChannelId;
+        // TODO: remove this check? should be preventable / never need to happen
+        const ledgerRecord = await Channel.forId(ledgerChannelId, txOrKnex);
+        if (!ledgerRecord) {
+          throw new Error('cannot fund app, no ledger channel w/ that asset. abort');
+        }
+        return ledgerChannelId;
+      }
+      throw new Error('ledger funded app has no fundingLedgerChannelId');
+    }
+    throw new Error('cannot fund app that is not ledger funded');
+  } else {
+    throw new Error('cannot fund unsupported app');
+  }
+};
```

Comment:
This entire function basically amount to `return channel.fundingLedgerChannelId`. Feel free to destroy it — it's mostly an artifact of history pre- #2760

Reply:
@tomclose would be nice to do away with this